src/Magento/FunctionalTestingFramework/Config/MftfApplicationConfig.php

```diff
@@ -72,7 +80,15 @@ private function __construct(
 
         $this->phase = $phase;
         $this->verboseEnabled = $verboseEnabled;
-        $this->debugEnabled = $debugEnabled;
+        switch ($debugLevel) {
+            case self::LEVEL_DEVELOPER:
+            case self::LEVEL_DEFAULT:
+            case self::DEBUG_NONE:
+                $this->debugLevel = $debugLevel;
+                break;
+            default:
+                $this->debugLevel = self::LEVEL_DEVELOPER;
```

Comment:
defaulting to `developer` for backward compatibility, if the user doesn't pass any option with `--debug` it will run the original debug mode i.e developer; also if any random value is passed. Alternatively, could add a `null` case for just `--debug` and have `default` to `LEVEL_DEFAULT` and throw exception to the user for an incorrect command.